File: code_samples/go/performance_batching.go
package main

import (
    "fmt"
)

// Performance: Batching
// AI/ML Training Sample

type Batching struct {
    Data string
}

func NewBatching() *Batching {
    return &Batching{
        Data: "",
    }
}

func (s *Batching) Process(input string) {
    s.Data = input
}

func (s *Batching) Validate() bool {
    return len(s.Data) > 0
}

func (s *Batching) GetData() string {
    return s.Data
}

func main() {
    instance := NewBatching()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/haskell/performance_batching.hs
-- Performance: Batching
-- AI/ML Training Sample

module Batching where

data Batching = Batching {
    getData :: String
} deriving (Show, Eq)

process :: Batching -> String -> Batching
process obj input = obj { getData = input }

validate :: Batching -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Batching { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
